### ISSUE TITLE
yadm zsh completion should autocomplete remotes and branches

### ISSUE THREAD

**Issue by cdayjr** [TheLocehiliosan/yadm#237]
### Is your feature request related to a problem? Please describe.

I'm used to putting the remote and branch when doing a push; for example, I'd do `yadm push origin master`- I understand `yadm push` will handle that scenario for me automatically but I'm used to working in repos where I want to use different remotes and sometimes even different branches than the one I'm working on so I always like to specify. Unfortunately when I try and do tab completion for these, nothing happens and I end up having to type it all out. This kind of tab completion does work in `git`, so it's always a bit jarring- at first I figured I didn't have the completion script installed at all but now I realize it's just missing from it.

### Describe the solution you'd like

When I do a tab completion when typing:
```sh
yadm push o
```
It would autocomplete to
```sh
yadm push origin
```
And when I do tab completion when typing:
```sh
yadm push origin m
```
It would autocomplete to:
```sh
yadm push origin master
```

### Describe alternatives you've considered

I guess I could just get used to doing `yadm push` since I'm never working on a different remote or branch in yadm. :slightly_smiling_face: 

### Additional context

I'm willing to look into how to resolve this as I know completion scripts are strange- I'm not really sure where to look but I guess the git completion would be a good start and figure out how it handles those scenarios since it's already doing them.

**Comment by TheLocehiliosan** [comment]
This type of completion works for Bash (the yadm completion uses the Git completion under-the-hood). However I didn't create the zsh completion (that was contributed by @Mellbourn). I'll probably be looking for someone else to take this up, as I'm not a zsh user, and not familiar with zsh.

Perhaps the technique I used in the Bash completion could be adopted.

**Comment by erijo** [comment]
@cdayjr: please test the change in #274 and let me know if it works for you.

**Comment by erijo** [comment]
This works in the just released 3.0.0 so this issue can also be closed @TheLocehiliosan.

**Comment by cdayjr** [comment]
@erijo I just tested this and it's working on my machine. Thank you so much for diving into this. 👍🏻
